innblue board Nrf9160 innblue22

olsky opened the pull request:
Dear all,

I would like to add one of our nrf91 boards to main.

please review and accept.
thank you

mbolivar commented:
Hi @olsky and thanks for your contribution!

You will however need to rebase this pull request into a sequence of individual commits without reverts or merges using an interactive git rebase to squash commits together, etc.

mbolivar commented:
@carlescufi I was looking for requirements saying "must be a line of commits, please squash things together" in https://docs.zephyrproject.org/latest/contribute/index.html#contribution-guidelines but it isn't explicitly spelled out there... am I missing something?

olsky commented:
thanks for the comments!

we tried again but same point... could you hint on: 
"Falling back to patching base and 3-way merge" --> why is this?

olsky commented:
the ADC test is failing, but it is not relevant in this case... as there are no definitions... could you please check? @carlescufi @mbolivar

mbolivar commented:
> as there are no definitions... could you please check? @carlescufi @mbolivar

If you look at the entire compiler output, you'll see this earlier:

```
/src/test_adc.c:179:2: error: #error "Unsupported board."
  179 | #error "Unsupported board."
      |  ^~~~~
```

Your board is getting built for this sample because the tests/drivers/adc/adc_api/testcase.yaml filters decided it supports adc, but there's no board support defined in that file.

mbolivar commented on boards/arm/nrf9160_innblue22/Kconfig.defconfig:
```diff
@@ -0,0 +1,57 @@
+	default "nrf9160_innblue22"
+
+config ADC_0
+	default y
```
That ADC error is likely because this is default y, but you'd need to look into the sanitycheck configuration for your board more to be sure.

https://docs.zephyrproject.org/latest/guides/test/sanitycheck.html

olsky replied:
ov! got your point -- thanks...

that is the only place for ADC and SPI... it is not used, so safe to remove.

Let me check it again.

mbolivar-nordic replied:
> that is the only place for ADC and SPI... it is not used, so safe to remove.

Are you sure it is OK to remove?

Just turning off ADC and SPI support to make the complaining test go away is not necessarily what should be done. The board porting guide has the requirements on what peripherals a board should be enabling:

https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations

In particular:

> Configure components that enable the use of these pins, such as configuring an SPI instance to use the usual Arduino SPI pins.

Please make sure to follow the guidelines, updating the tests as needed to pass, instead of just silencing test errors.

mbolivar-nordic replied:
> > Configure components that enable the use of these pins, such as configuring an SPI instance to use the usual Arduino SPI pins.
> 
> Please make sure to follow the guidelines, updating the tests as needed to pass, instead of just silencing test errors.

I'm not sure whether this has been explicitly addressed. It needs to be.

bwasim replied:
@mbolivar , The innblue 2.2 board doesn't have support for ADC / SPI. As we used the nRF9160 dev kit as a baseline for our board, these definitions renamed in place. Now we've decided to remove them altogether. 

In short, we don't want to fix the application build errors for our board because it doesn't have supported for the mentioned peripherals.

bwasim commented:
@lemrey / @ioannisg / @carlescufi - Can anyone please take a look at this pull request and provide feedback ? This is the addition of a custom board based on the nRF91 SoC. As can be seen, all integration tests are already passing..

olsky commented:
@carlescufi could we push this version too?

ioannisg commented:
> @lemrey / @ioannisg / @carlescufi - Can anyone please take a look at this pull request and provide feedback ? This is the addition of a custom board based on the nRF91 SoC. As can be seen, all integration tests are already passing..

Have you pushed an .rst document for the board documentation? I could only see the board image...

ioannisg reviewed (commented):
Please, push an .rst document for board documentation :)

carlescufi commented:
> @carlescufi could we push this version too?

As @ioannisg mentioned, please add an .rst to the documentation just like all other boards.

carlescufi commented:
> @carlescufi I was looking for requirements saying "must be a line of commits, please squash things together" in https://docs.zephyrproject.org/latest/contribute/index.html#contribution-guidelines but it isn't explicitly spelled out there... am I missing something?

@mbolivar-nordic all we have is:
```
One general practice we encourage, is to make small, controlled changes. This practice simplifies review, makes merging and rebasing easier, and keeps the change history clear and clean.
```
in https://docs.zephyrproject.org/latest/contribute/index.html#contribution-workflow

olsky commented:
> Please, push an .rst document for board documentation :)

thanks for reminder! pushed...

mbolivar-nordic reviewed (changes requested):
We need a real email address, not `Signed-off-by: Oleh Lozynskyy <olsky@users.noreply.github.com>` type addresses. Any commits with the noreply emails were merged by mistake.

olsky commented:
> We need a real email address, not `Signed-off-by: Oleh Lozynskyy <olsky@users.noreply.github.com>` type addresses. Any commits with the noreply emails were merged by mistake.

ov, yepp, let me change the default signature.

bwasim commented:
@reviewers, All requested changes are addressed in the pull request. Please review..

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,106 @@
+
+Following parts are available:
+
+* Accelerometer      > ST LIS2DH12
+* CryptoElement      > Atmel ATECC608a
+* Humidity Sensor    > ST HTS221
+* Qi charger         > TI BQ51013
+* Battery guel gauge > TI BQ27421
+
```
```suggestion
* Accelerometer: ST LIS2DH12
* CryptoElement: Atmel ATECC608a
* Humidity Sensor: ST HTS221
* Qi charger: TI BQ51013
* Battery guel gauge: TI BQ27421
```

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,106 @@
+  peripheral address space) is defined to be secure accessible only.
+- Secure boot.
+
+Building Secure/Non-Secure Zephyr applications
```
Please follow the doc/templates/board.tmpl structure throughout, including the "Programming and Debugging" section. The rendered documentation should have exact west commands users can copy/paste to build and flash (if flashing and debugging with west is not possible, then board.cmake has problems). You can see how this is done for nRF9160-DK using `zephyr-app-commands`.

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,106 @@
+Hardware
+********
+
+Following parts are available:
```
```suggestion
The following parts are built into the board:
```

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,106 @@
+Overview
+********
+
+The "innblue v2.2" is a cellular IoT sensor development board, which
```
The page uses "INNBLUE22", "innblue v2.2", and "innblue22" as the board's name. It would be nice to pick one and be consistent. I see that nRF9160-DK is guilty of the same offense in its documentation, so this is a non-blocking request. I'll send a PR fixing the DK documentation next.

mbolivar-nordic replied:
> I'll send a PR fixing the DK documentation next.

https://github.com/zephyrproject-rtos/zephyr/pull/24002

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/nrf9160_innblue22_common.dts:
```diff
@@ -0,0 +1,219 @@
+};
+
+/ {
+	/* SRAM allocated and used by the BSD library */
```
@anangl @ioannisg is this something we should move to an include file at some point?

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/Kconfig.defconfig:
```diff
@@ -0,0 +1,50 @@
+config BOARD
+	default "nrf9160_innblue22"
+
+config I2C_2
```
Please be aware you may need to revisit this if #19480 is merged first.

mbolivar-nordic reviewed (commented):
I looked at the board docs and DTS. Not supporting ADC seems fine since there's no Arduino connector or anything like that. A few docs improvements requested.

bwasim commented:
@all, please re-review.. All tests are passing now..

mbolivar-nordic commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,145 @@
+Overview
+********
+
+The "nRF9160_innblue22" is a cellular IoT sensor development board, which
```
```suggestion
The nRF9160 innblue22 is a cellular IoT sensor development board, which
```

This looks a lot like the board name you would type into the console (`nrf9160_innblue22`), which is case sensitive. So I suggest removing the underscore and quotes to keep it consistent with the page title and avoid confusion with the command-line board name.

mbolivar-nordic reviewed (approved):
Docs look fine now, thanks for your patience. I have one suggestion for you to consider.

These boards look very nice. Thanks for taking the time to contribute Zephyr support for them.

anangl commented on boards/arm/nrf9160_innblue22/nrf9160_innblue22_common.dts:
```diff
@@ -0,0 +1,219 @@
+		compatible = "gpio-keys";
+
+		button0: button_0 {
+			gpios = <&gpio0 31 GPIO_PUD_PULL_UP>;
```
Use `(GPIO_PULL_UP | GPIO_ACTIVE_LOW)` here.
`GPIO_PUD_PULL_UP` is deprecated.

anangl commented on boards/arm/nrf9160_innblue22/nrf9160_innblue22_common.dts:
```diff
@@ -0,0 +1,219 @@
+
+&pwm0 {
+	status = "okay";
+	ch0-pin = <5>;
+	ch1-pin = <6>;
+	ch2-pin = <7>;
+	ch0-inverted;
+	ch1-inverted;
+	ch2-inverted;
```
The LEDs are active high, right? In that case the PWM channels for them should not be inverted.

olsky replied:
on this board LEDs are active low

anangl replied:
You should then add `GPIO_ACTIVE_LOW` flags to their [definitions](https://github.com/zephyrproject-rtos/zephyr/pull/23451/files/35034b83d93ae61fd0d27563c4bff0a13115455b#diff-2991b9170f6f938500bb04b16084b037R21) in dts.

anangl commented on boards/arm/nrf9160_innblue22/nrf9160_innblue22_defconfig:
```diff
@@ -0,0 +1,24 @@
+
+# Enable uart driver
+CONFIG_SERIAL=y
+CONFIG_UART_NRFX=y
```
This is redundant. `UART_NRFX` is by default enabled for all nRF SoCs, when `SERIAL` is enabled.
Same story in `boards/arm/nrf9160_innblue22/nrf9160_innblue22ns_defconfig`.

olsky replied:
thank you, good point!

anangl commented on boards/arm/nrf9160_innblue22/nrf9160_innblue22_common.dts:
```diff
@@ -0,0 +1,219 @@
+	hts221@5f {
+		compatible = "st,hts221";
+		reg = <0x5f>;
+		drdy-gpios = <&gpio0 3 GPIO_INT_ACTIVE_HIGH>;
```
`GPIO_INT_ACTIVE_HIGH` is deprecated. Use `GPIO_ACTIVE_HIGH` instead.

anangl commented on boards/arm/nrf9160_innblue22/nrf9160_innblue22_common.dts:
```diff
@@ -0,0 +1,219 @@
+	lis2dh12-accel@19 {
+		compatible = "st,lis2dh";
+		reg = <0x19>;
+		irq-gpios = <&gpio0 0 0>, <&gpio0 1 0>;
```
Use `GPIO_ACTIVE_HIGH` as flags (the ending zeros) for both these GPIOs, for consistency with other uses the sensor in the Zephyr tree.

olsky commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,145 @@
+* CryptoElement: Atmel ATECC608a
+* Humidity Sensor: ST HTS221
+* Qi charger: TI BQ51013
+* Battery guel gauge: TI BQ27421
```
aaa :))) yes, fuel gauge :) thanks!

olsky commented on boards/arm/nrf9160_innblue22/innblue22_board_init.c:
```diff
@@ -0,0 +1,41 @@
+	return (err);
+}
+
+DEVICE_INIT(vdd_5v0_pwr_ctrl_init, "", pwr_ctrl_init, NULL, NULL,
```
good question... got a hint on slack and used this in this form (copied from one of nordic boards init)...

for me was "device" as "board" init...
what are the benefits to use sys_init? and what the main differences? 
Will it behave same in "secure" and "non secure" as well?
Thank you in advance!

anangl replied:
Both ways it will be effectively the same, as both these macros call `DEVICE_AND_API_INIT()` underneath. The thing is that a call to `DEVICE_INIT()` with an empty device name looks a bit artificial, and `SYS_INIT()` is a dedicated macro for such initialization routines.

bwasim committed:
boards/arm: Adding support for nRF9160 InnBlue22 Board.

Adding support for nRF9160 based InnBlue board V2.2. Supports both
Secure and Non-Secure configurations along with various sensors
(lis2dh12 / hts221) and devices(i2c / pwm).

Signed-off-by: Bilal Wasim <bilalwasim676@gmail.com>

bwasim commented:
@galak , I don't understand why so many reviewers are added automatically as I made an update to this pull request. Something wrong with the infrastructure ?

nandojve commented:
@bwasim probably there is a problem, I just tried remove myself.

pabigot commented on boards/arm/nrf9160_innblue22/doc/index.rst:
```diff
@@ -0,0 +1,145 @@
+********
+
+The nRF9160 innblue22 is a cellular IoT sensor development board, which
+is based on the nRF9160 SiP, and features NB-IoT and LTE-M connectivity.
```
It would be nice if this included information about who makes it and a product information page, so interested parties can get one.

If this has not been released, please add that as soon as it's available.  (google isn't finding anything relevant about this device.)

olsky replied:
We are still busy with the info page and marketing... I will update as soon as it will be ready. Last few weeks were a real slowdown...
thank you for your feedback!

olsky commented:
heya, thank you! Party! :o)
